### ISSUE TITLE
Quay hosted CNI ARM image unavailable

### ISSUE THREAD

**Issue by surajssd** [projectcalico/calico#4692]
## Expected Behavior

Since this issue #4227 was fixed. I wanted to adpot the quay based calico images. But there is no CNI ARM image available on Quay. All other images work fine except for the CNI one.

## Current Behavior

On an ARM machine:

```console
# uname -a
Linux sv-c1-large-arm-01 5.4.0-40-generic #44-Ubuntu SMP Mon Jun 22 23:59:48 UTC 2020 aarch64 aarch64 aarch64 GNU/Linux
```

When I pull the quay CNI image it fails as:

```
# docker run quay.io/calico/cni:v3.18.1
Unable to find image 'quay.io/calico/cni:v3.18.1' locally
v3.18.1: Pulling from calico/cni
a34202e9311d: Pull complete 
308006a25eb5: Pull complete 
3fafc2cf7ef2: Pull complete 
Digest: sha256:b946fb6149b0dce3324d86eadf2d98ded4931e254d261a92eacf3a5d8441708d
Status: Downloaded newer image for quay.io/calico/cni:v3.18.1
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error
```

## Possible Solution

Upload ARM image for CNI like it is done for other calico components.

**Comment by caseydavenport** [comment]
We don't yet push multi-arch manifests to quay.io, since for a long time it did not support them. If you're pulling from quay you probably need to specify the arch explicitly: `docker pull quay.io/calico/cni:v3.18.1-arm64`

**Comment by surajssd** [comment]
Thanks for the update @caseydavenport !

**Comment by markus-seidl** [comment]
Also stumbled on this problem, maybe you can mention it in the manual (until it's fixed).

**Comment by andybrook** [comment]
Based on my write up on a related ticket here - https://github.com/kubernetes-sigs/kubespray/issues/7934#issuecomment-992886090 I don't believe this is only an issue for the CNI image.

I checked for v3.21.2 which was pushed to docker hub only 8 days go and confirmed it's still an issue for recent releases.

This is currently causing some issues with multi-arch clusters in kubespray, I appreciate this is a little niche, but it would be great if both repos could be supported equally and multi-arch has been supported by quay.io for some time now.

**Comment by caseydavenport** [comment]
This PR went in to our build machinery which should trigger future releases to push multi-arch images to quay: https://github.com/projectcalico/go-build/pull/351

It looks like v3.21 was built with the v0.59 version of our build tooling, which means it didn't pick up this change. I'm expecting this to come through in v3.22 at the latest.

It is also possible we could bump the go-build version for the v3.21 branch to v0.63 and then future v3.21 patch releases will have this change.

**Comment by caseydavenport** [comment]
This PR went in to our build machinery which should trigger future releases to push multi-arch images to quay: https://github.com/projectcalico/go-build/pull/351

It looks like v3.21 was built with the v0.59 version of our build tooling, which means it didn't pick up this change. I'm expecting this to come through in v3.22 at the latest.

It is also possible we could bump the go-build version for the v3.21 branch to v0.63 and then future v3.21 patch releases will have this change.

**Comment by frozenprocess** [comment]
Fixed in v3.21.3+
```
docker manifest inspect quay.io/calico/cni:release-v3.21
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 946,
         "digest": "sha256:2fc54a8e3cb84c76f426c4b4ba9751b1b06f9dcbfa3d6acfaebb1363edcc6bd8",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1369,
         "digest": "sha256:fa29d367597f2146ead38d8d4b686928569772d883da859e121bdeb0b5aec63e",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 946,
         "digest": "sha256:ca5fc085a81ab94d82f2f47ca48c921782ec3271362b99e830bd42fbd33be4f3",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 946,
         "digest": "sha256:25a15fa9b307c9eedd4e84888db252974eabd39bd2dee062bca7eed303e4da81",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      }
   ]
}
```
